util/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/monitoring/MonitoringService.java

```diff
@@ -0,0 +1,191 @@
+package com.google.cloud.healthcare.imaging.dicomadapter.monitoring;
+
+import com.google.api.Metric;
+import com.google.api.MetricDescriptor;
+import com.google.api.MonitoredResource;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.cloud.healthcare.imaging.dicomadapter.GcpMetadataUtil;
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.monitoring.v3.CreateTimeSeriesRequest;
+import com.google.monitoring.v3.Point;
+import com.google.monitoring.v3.ProjectName;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TimeSeries;
+import com.google.monitoring.v3.TypedValue;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MonitoringService {
+
+  private static final int DELAY = 60;
+
+  private static final String META_PROJECT_ID = "project/project-id";
+  private static final String META_LOCATION = "instance/zone";
+  private static final String META_CLUSTER_NAME = "instance/attributes/cluster-name";
+
+  private static Logger log = LoggerFactory.getLogger(MonitoringService.class);
+  private static MonitoringService INSTANCE;
+  private static boolean ENABLED = true;
+
+  private final MetricServiceClient client;
+  private final ScheduledExecutorService service;
+  private final HashMap<IMonitoringEvent, Long> aggregateEvents;
+  private final IMonitoringEvent[] monitoredEvents;
+  private final MonitoredResource monitoredResource;
+
+  private String projectId;
+
+  private MonitoringService(String projectId, IMonitoringEvent[] monitoredEvents,
+      HttpRequestFactory requestFactory) throws IOException {
+    client = MetricServiceClient.create();
+
+    aggregateEvents = new HashMap<>();
+
+    if (projectId != null && projectId.length() > 0) {
```

Comment:
Now that we require projectId to be specified, I think we can remove this code and the META_PROJECT_ID constant.

Reply:
Ok